engine/src/main/java/org/hibernate/search/spatial/SpatialFieldBridgeByHash.java

```diff
@@ -63,6 +65,16 @@ public void configureFieldMetadata(String name, FieldMetadataBuilder builder) {
 		}
 	}
 
+	@Override
+	protected void initializeIndexedFieldNames(String fieldName) {
+		super.initializeIndexedFieldNames( fieldName );
+
+		hashIndexedFieldNames = new String[bottomSpatialHashLevel + 1];
```

Comment:
This seems to create the hash field names also if not needed (in case of numericFieldsIndex).